src/utils/batch-operate/batch-util.ts

```diff
@@ -0,0 +1,53 @@
+import type { AirtopClient } from "wrapper/AirtopClient";
+import type {
+	BatchOperateConfig,
+	BatchOperationInput,
+	BatchOperationResponse,
+	BatchOperationUrl,
+} from "./types";
+import { EventEmitter } from "node:events";
+import { distributeUrlsToBatches } from "./helpers";
+import { SessionQueue } from "./SessionQueue";
+
+const DEFAULT_MAX_WINDOWS_PER_SESSION = 1;
+const DEFAULT_MAX_CONCURRENT_SESSIONS = 30;
+
+export const batchOperate = async (
```

Comment:
I think we want users to be able to register their own listeners for the events we're emitting. We might need to restructure this just a little bit to make this happen. Maybe we can catch up on this one, it may not need to happen immediately. 

I was thinking about supporting an interface like the following from the client-side to make this happen:

```typescript
// If they want to add their own event handlers
const batchOperator = airtopClient.createBatchOperator(...);
batchOperator.on('foo', () => { /* handle foo */});
await batchOperator.run();

// If they don't care about events they can use this function directly
airtopClient.batchOperate(...);
```

I don't think it would require too many changes from your current code here to make this happen if we want to do this in the short term. We could implement this later if we continue to support the same signature on `batchOperate()` to avoid any breaking changes. cc/ @nrotstan for thoughts on this.

Reply:
I agree that it makes sense to let clients register listeners and receive events as items are processed (and other relevant events).

Reply:
See comment above (sounds like we might want to address this in a follow-up)